cmd/gazelle/integration_test.go

```diff
@@ -1164,6 +1164,98 @@ go_library(
 	})
 }
 
+func TestUpdateReposWithQueryToWorkspace(t *testing.T) {
+	files := []testtools.FileSpec{
+		{
+			Path: "WORKSPACE",
+			Content: `
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "io_bazel_rules_go",
+    urls = [
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/rules_go/releases/download/0.19.1/rules_go-0.19.1.tar.gz",
+        "https://github.com/bazelbuild/rules_go/releases/download/0.19.1/rules_go-0.19.1.tar.gz",
+    ],
+    sha256 = "8df59f11fb697743cbb3f26cfb8750395f30471e9eabde0d174c3aebc7a1cd39",
+)
+
+http_archive(
+    name = "bazel_gazelle",
+    urls = [
+        "https://storage.googleapis.com/bazel-mirror/github.com/bazelbuild/bazel-gazelle/releases/download/0.18.1/bazel-gazelle-0.18.1.tar.gz",
+        "https://github.com/bazelbuild/bazel-gazelle/releases/download/0.18.1/bazel-gazelle-0.18.1.tar.gz",
+    ],
+    sha256 = "be9296bfd64882e3c08e3283c58fcb461fa6dd3c171764fcc4cf322f60615a9b",
+)
+
+load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+
+go_rules_dependencies()
+
+go_register_toolchains(nogo = "@bazel_gazelle//:nogo")
+
+load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
+
+gazelle_dependencies()
+`,
+		},
+	}
+
+	dir, cleanup := testtools.CreateFiles(t, files)
+	defer cleanup()
+
+	os.Setenv("GOCACHE", os.TempDir())
+	os.Setenv("GOPATH", os.TempDir())
```

Comment:
`os.TempDir()` will return something like `/tmp`. We can't dump files there without deleting them. Create directories with `io/ioutil.TestDir` instead and delete them with `os.RemoveAll`.

Also, these variables are global and will affect other tests. That's probably okay: other tests will need these in the future. However, these should be set in a `TestMain` function. Make sure to clean up the directories, even if `m.Run` panics.

One more thing: the windows tests are failing because `go.exe` is not in `PATH`. Gazelle will look in `GOROOT/bin`, so let's so GOROOT. Add `data = ["@go_sdk//:files"],` to the `go_test` rule `//cmd/gazelle:go_default_test` to make sure the go tool and all the related files are visible in the sandbox. Add `args = ["-goroot=$(location @go_sdk//:ROOT)"],` to the same rule. Define a `flag.String` for `goroot` in the test. In the `TestMain`, call `flag.Parse` and set the `GOROOT` environment variable to the `filepath.Dir` of that flag.